mock_tests/test_connection.py

```diff
@@ -0,0 +1,32 @@
+import json
+from typing import Dict
+
+import pytest
+from werkzeug import Request, Response
+
+import weaviate
+
+
+@pytest.mark.parametrize(
+    "header",
+    [
+        {},
+        {"Authorization": "Bearer test"},
+        {"Authorization": "Bearer test", "SomethingElse": "Value"},
```

Comment:
It should be case in-sensitive, so maybe you can add some tests to test for that too

Reply:
Hmm, this test only makes sure that what you set as additional headers shows up in the actual http-header. Which part should be case-insensitive?

Reply:
the `key` (this is true for all HTTP headers): https://github.com/semi-technologies/weaviate-python-client/blob/main/weaviate/connect/connection.py#L85-L86

Reply:
added an additional test with lowercase authentication